tasks/task_content.c

```diff
@@ -1686,6 +1686,22 @@ static void task_push_to_history_list(
 
             content_get_subsystem_friendly_name(path_get(RARCH_PATH_SUBSYSTEM), subsystem_name, sizeof(subsystem_name));
 
+#ifdef HAVE_ENTRYSTATES
+            char state[PATH_MAX_LENGTH];
```

Comment:
Variable declarations need to go at the top of the scope for C89 compatibility

Reply:
I'm sure you know about -Wdeclaration-after-statement -Werror=declaration-after-statement. Why not add those to the Makefile?